tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -5380,20 +5380,32 @@ class HLSLExternalSource : public ExternalSemaSource {
       }
 
       QualType ArgTy = Arg.getAsType();
+      // Allow DependentType for case like
+      // template <typename T> void foo(GroupNodeInputRecords<T> data) {}
+      if (ArgTy->isDependentType())
```

Comment:
Is the `case AR_TOBJ_DEPENDENT` in `DiagnoseNodeStructArgument` still hit with this code?  If so, in what case?  Is there a test here that hits it?

Reply:
Added.